Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -1003,11 +1003,18 @@ void LineLayout::shiftLinesBy(LayoutUnit blockShift)
     for (auto& line : m_inlineContent->displayContent().lines)
         line.moveInBlockDirection(blockShift, isHorizontalWritingMode);
 
-    for (auto& box : m_inlineContent->displayContent().boxes) {
+    LayoutUnit dx = (isHorizontalWritingMode) ? 0_lu : blockShift;
+    LayoutUnit dy = (isHorizontalWritingMode) ? blockShift : 0_lu;
+    for (InlineDisplay::Box& box : m_inlineContent->displayContent().boxes) {
```

Comment:
(while I understanding the reason)This change is not inline with WebKit's usage of auto.

Reply:
(if you would like to change the way WebKit uses auto, please start a discussion at webkit-dev)

Reply:
Where is WebKit's guidelines for the use of auto documented? Because it's not in https://webkit.org/code-style-guidelines/

Reply:
(And yes, I can change it over, but this function is incomprehensible without explicit typing.)

Reply:
There was some prior discussion on the webkit-dev mailing list

Reply:
I don't think I said guideline, I meant more of a general direction based on prior discussions.

Reply:
This was relatively recent discussion (2022-2021?), so probably no-one has had time to properly write it down.

Reply:
Ok, I'll go poking around for that thread at some point.